### ISSUE TITLE
[Internal Rule]: no-unused-messageId

### ISSUE THREAD

**Issue by anikethsaha** [eslint/eslint#13221]
**Please describe what the rule should do:**
It should report whenever there is a message id defined in rule's meta property but its not been used anywhere in the rule's definition

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
function checkMetaMessages(context, exportsNode) {
    if (!messages) {
        context.report({
            node: exportsNode,
            messageId: "used"
        });
      
    }
}


module.exports = {
    meta: {
        docs: {
            description: "",
            category: "Internal",
            recommended: false
        },
        schema: [],
        type: "suggestion",
        messages: {
          used: "Expected `meta.messages` property.",
          notUsed: "Expected `meta.messages` property."
      // ^^^ report here that this message id is not being used
        }
    },

    create(context) {
        return {
            "Program"(node) {
                checkMetaMessages(context, node);
            }
          
        };
    }
};

```

**Why should this rule be included in ESLint (instead of a plugin)?**
Internal rule for eslint

**Are you willing to submit a pull request to implement this rule?**
Yes, 
[rough implementation](https://astexplorer.net/#/gist/64119003e065cda0b75db75bb6db98ff/c857d241d8d87beb211603c47588a32d2ae1ba61)

**Comment by mdjermanovic** [comment]
That would be great, but would it be possible to do this without false positives?

There are some rules where it's more complex than `context.report({ messageId: "..." })`, for example [arrow-body-style](https://github.com/eslint/eslint/blob/fdfa364c1b0d88689d02cbe8ae848a3e323209aa/lib/rules/arrow-body-style.js#L123) or [generator-star-spacing](https://github.com/eslint/eslint/blob/fdfa364c1b0d88689d02cbe8ae848a3e323209aa/lib/rules/generator-star-spacing.js#L152).

**Comment by g-plane** [comment]
I think those message ids which can't be checked statically can be skipped, and most of rules are still able to be checked.

**Comment by anikethsaha** [comment]
I would prefer to skip them. 

though we can check all `Literal` and check for their `value`. this would solve the `arrow-body-style` but not for  `generator-star-spacing` so I would prefer to skip for this rule may be by disabling this rule ?

WDYT ?

**Comment by g-plane** [comment]
Another suggestion: implement this rule into [eslint-plugin-eslint-plugin](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin).

**Comment by anikethsaha** [comment]
> Another suggestion: implement this rule into [eslint-plugin-eslint-plugin](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin).

yeah, this seems good as well 👍 .

**Comment by aladdin-add** [comment]
it has been accepted in https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/57

PRs are welcome! 😄
